pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -882,7 +975,7 @@ public MemoryLimitController getMemoryLimitController() {
             }
             schema = schema.clone();
             if (schema.requireFetchingSchemaInfo()) {
-                Schema finalSchema = schema;
+                @SuppressWarnings("rawtypes") Schema finalSchema = schema;
```

Comment:
This should work too:

```suggestion
                Schema<?> finalSchema = schema;
```